### ISSUE TITLE
Address accessibility

### ISSUE THREAD

**Issue by carawarner** [carawarner/pantheon-demo#42]
[Source](https://www.wuhcag.com/wcag-checklist/).

**Legend**

- [x] Did it
- [ ] Did not do it (yet)
- [x] ~~Not applicable to the current pantheon demo~~

# WCAG 2.0 checklist Level A (Beginner)

- [x] 1.1.1 – ~~Non-text Content | Provide text alternatives for non-text content~~
- [x] 1.2.1 – ~~Audio-only and Video-only (Pre-recorded) | Provide an alternative to video-only and audio-only content~~
- [x] 1.2.2 – ~~Captions (Pre-recorded) | Provide captions for videos with audio~~
- [x] 1.2.3 – ~~Audio Description or Media Alternative (Pre-recorded) | Video with audio has a second alternative~~
- [x] 1.3.1 – Info and Relationships | Logical structure
- [x] 1.3.2 – Meaningful Sequence | Present content in a meaningful order
- [ ] 1.3.3 – Sensory Characteristics | Use more than one sense for instructions
- [x] 1.4.1 – Use of Colour | Don’t use presentation that relies solely on colour
- [x] 1.4.2 – ~~Audio Control | Don’t play audio automatically~~
- [x] 2.1.1 – Keyboard | Accessible by keyboard only
- [x] 2.1.2 – No Keyboard Trap | Don’t trap keyboard users
- [x] 2.2.1 – ~~Timing Adjustable | Time limits have user controls~~
- [x] 2.2.2 – ~~Pause, Stop, Hide | Provide user controls for moving content~~
- [x] 2.3.1 – ~~Three Flashes or Below | No content flashes more than three times per second~~
- [x] 2.4.1 – ~~Bypass Blocks | Provide a ‘Skip to Content’ link~~
- [x] 2.4.2 – ~~Page Titled | Use helpful and clear page titles~~ - inapplicable because the recommended "Page name – Page description – Website name" format would produce "Pantheon Generator - Pantheon Generator - Pantheon Generator"
- [x] 2.4.3 – Focus Order | Logical order
- [ ] 2.4.4 – Link Purpose (In Context) | Every link’s purpose is clear from its context
- [x] 3.1.1 – Language of Page | Page has a language assigned
- [x] 3.2.1 – On Focus | Elements do not change when they receive focus
- [x] 3.2.2 – On Input | Elements do not change when they receive input
- [ ] 3.3.1 – Error Identification | Clearly identify input errors
- [ ] 3.3.2 – Labels or Instructions | Label elements and give instructions
- [x] 4.1.1 – Parsing | No major code errors
- [ ] 4.1.2 – Name, Role, Value | Build all elements for accessibility

# WCAG 2.0 checklist Level AA (Intermediate)
- [x] 1.2.4 – ~~Captions (Live) | Live videos have captions~~
- [x] 1.2.5 – ~~Audio Description (Pre-recorded) | Users have access to audio description for video content~~
- [ ] 1.4.3 – Contrast (Minimum) | Contrast ratio between text and background is at least 4.5:1
- [x] 1.4.4 – Resize Text | Text can be resized to 200% without loss of content or function
- [x] 1.4.5 – ~~Images of Text | Don’t use images of text~~
- [x] 2.4.5 – ~~Multiple Ways | Offer several ways to find pages~~
- [x] 2.4.6 – Headings and Labels | Use clear headings and labels
- [x] 2.4.7 – Focus Visible | Ensure keyboard focus is visible and clear
- [x] 3.1.2 – ~~Language of Parts | Tell users when the language on a page changes~~
- [x] 3.2.3 – ~~Consistent Navigation | Use menus consistently~~
- [x] 3.2.4 – Consistent Identification | Use icons and buttons consistently
- [ ] 3.3.3 – Error Suggestion | Suggest fixes when users make errors
- [ ] 3.3.4- Error Prevention (Legal, Financial, Data) | Reduce the risk of input errors for sensitive data

# WCAG 2.0 checklist Level AAA (Advanced)

- [x] 1.2.6 – ~~Sign Language (Pre-recorded) | Provide sign language translations for videos~~
- [x] 1.2.7 – ~~Extended Audio description (Pre-recorded) | Provide extended audio description for videos~~
- [x] 1.2.8 – ~~Media Alternative (Pre-recorded) | Provide a text alternative to videos~~
- [x] 1.2.9 – ~~Audio Only (Live) | Provide alternatives for live audio~~
- [ ] 1.4.6 – Contrast (Enhanced) | Contrast ratio between text and background is at least 7:1
- [x] 1.4.7 – ~~Low or No Background Audio | Audio is clear for listeners to hear~~
- [ ] 1.4.8 – Visual Presentation | Offer users a range of presentation options
- [x] 1.4.9 – ~~Images of Text (No Exception) | Don’t use images of text~~
- [x] 2.1.3 – Keyboard (No Exception) | Accessible by keyboard only, without exception
- [x] 2.2.3 – ~~No Timing | No time limits~~
- [x] 2.2.4 – Interruptions | Don’t interrupt users
- [x] 2.2.5 – ~~Re-authenticating | Save user data when re-authenticating~~
- [x] 2.3.2 – ~~Three Flashes | No content flashes more than three times per second~~
- [x] 2.4.8 – ~~Location | Let users know where they are~~
- [ ] 2.4.9 – Link Purpose (Link Only) | Every link’s purpose is clear from its text
- [x] 2.4.10 – Section Headings | Break up content with headings
- [ ] 3.1.3 – Unusual words | Explain any strange words
- [x] 3.1.4 – ~~Abbreviations | Explain any abbreviations~~
- [ ] 3.1.5 – Reading Level | Users with nine years of school can read your content
- [ ] 3.1.6 – Pronunciation | Explain any words that are hard to pronounce
- [x] 3.2.5 – ~~Change on Request | Don’t change elements on your website until users ask~~
- [ ] 3.3.5 – Help | Provide detailed help and instructions
- [ ] 3.3.6 – Error Prevention (All) | Reduce the risk of all input errors

**Comment by carawarner** [comment]
# Notes

I verified you can use the Pantheon Generator without a mouse. Tabbing and arrow keys can be used to select and manipulate the inputs for chromosomes, gender, and seed words.


# TODO

- [ ] Provide visual indicator that an input is selected

**Comment by carawarner** [comment]
# TODOs

- [ ] Visual queue to indicate which input is selected for those using keyboard only and tab-to-select.

**Comment by carawarner** [comment]
# Form Improvements 

- [ ] Gray out and disable button during generation
- [ ] Update the message at the top of the results section to include references to the parents of creation. 
- [ ] Add instructions above the form.
- [ ] Add form validation on seed words: they must be filled in and if they aren't there should be an error message.
- [ ] Add form validation on gender: they must be selected and if they aren't there should be an error message.
- [ ] Add form validation on chromosomes: there must be an XX and an XY and if there isn't there should be an error message.
- [ ] Add more form validation on seed word input: it should be a string of alpha-only characters with no whitespace, and if it isn't there should be an error message.

**Comment by carawarner** [comment]
Update: I am satisfied that I've gotten close enough for v1.0.
